src/crypto/chacha.h

```diff
@@ -69,22 +69,26 @@ namespace crypto {
     chacha20(data, length, key.data(), reinterpret_cast<const uint8_t*>(&iv), cipher);
   }
 
-  inline void generate_chacha_key(const void *data, size_t size, chacha_key& key) {
+  inline void generate_chacha_key(const void *data, size_t size, chacha_key& key, uint64_t kdf_rounds) {
     static_assert(sizeof(chacha_key) <= sizeof(hash), "Size of hash must be at least that of chacha_key");
     tools::scrubbed_arr<char, HASH_SIZE> pwd_hash;
     crypto::cn_slow_hash(data, size, pwd_hash.data(), 0/*light*/, 0/*variant*/, 0/*prehashed*/);
+    for (uint64_t n = 1; n < kdf_rounds; ++n)
+      crypto::cn_slow_hash(pwd_hash.data(), pwd_hash.size(), 0/*light*/, pwd_hash.data(), 0/*variant*/, 0/*prehashed*/);
```

Comment:
This is a wrong patch which is later fixed in 74757a23980fa484792b4997e48635f2eadd229e. It should have not been introduced here in the first place.

Reply:
You are referencing the wrong commit in the highlighted lines. Please see c2fa3e1fc instead. This is also 4103 which is before your references 4131.

Reply:
I'm not sure what you mean, but the very suggested commit https://github.com/aeonix/aeon/commit/c2fa3e1fc072c069fff5f251ef5a6edcffb6754c is the one that contains this wrong patch.